### PR TITLE
Buttons are disabled when there is a null in the binding chain for Command

### LINKED ISSUE #846: Automatically disable buttons/menu items when bound command is not found.
Sometimes it's useful to bind a button/menu item to a command that may not be present in the current view, e.g.:

```
<Window>
    <Menu>
        <MenuItem Header="Save" Command="{Binding CurrentDocument.Save}"/>
    </Menu>
</Window>

```

In this case, when `CurrentDocument` is null, the Save menu item needs to be disabled.

It should be possible to disable the control automatically when a binding error is received on the `Command` property.

### PR DESCRIPTION
**Pull request by jkoritzinsky**
Fixes #846.

### CODE REVIEW

File: src/Avalonia.Controls/Button.cs [hunk 1]
```diff
@@ -41,8 +42,9 @@ public class Button : ContentControl
         /// <summary>
         /// Defines the <see cref="Command"/> property.
         /// </summary>
-        public static readonly StyledProperty<ICommand> CommandProperty =
-            AvaloniaProperty.Register<Button, ICommand>(nameof(Command));
+        public static readonly DirectProperty<Button, ICommand> CommandProperty =
```
Comment (kekekeks) [inline src/Avalonia.Controls/Button.cs]: If I remember correctly, direct properties lack some features. So It's probably better to subscribe to property change notification here.

Comment (jkoritzinsky) [inline src/Avalonia.Controls/Button.cs]: The subscription to the property changed notification happens in the static constructor. It still works with DirectProperties

**Review by grokys — CHANGES_REQUESTED** [review]
I tried adding [a few unit tests](https://gist.github.com/grokys/1ddb942f8a0a49090eef59ec0349e452) to test this and found some problems.

The first problem is caused by [this code](https://github.com/AvaloniaUI/Avalonia/blob/e03187073879bd19d54959ca0411b2f4d322a139/src/Markup/Avalonia.Markup/Data/ExpressionObserver.cs#L242-L252) which results in the binding error not being notified on null. I think we probably need to remove this code - for the moment that will spew a load of unwanted warnings but I think we need to address that a different way in another PR.

The second problem is that if the `Command` is updated to point to a disabled `ICommand` then the button will be enabled even when it should be disabled.

File: Avalonia.sln [hunk 2]
```diff
@@ -1,5 +1,6 @@
 Microsoft Visual Studio Solution File, Format Version 12.00
 # Visual Studio 15
+VisualStudioVersion = 15.0.26403.3
```
Comment (kekekeks) [inline Avalonia.sln]: Duplicate

**Comment by grokys** [comment]
@jkoritzinsky thanks for updating - I'm not ignoring this but I want to do a proper review and merge it after we've released alpha 5 if that's ok?

**Comment by jkoritzinsky** [comment]
Sure!

**Review by grokys — APPROVED** [review]

**Pull request merged**